File: katas/7 kyu/isograms/solution.py
import codewars_test as test
import re
# -------------------------------- MY SOLUTION ------------------------------- #
def is_isogram(string):
  if string == "":
    return True
  for x in string.lower() :
    if string.lower().count(x.lower()) > 1 or x.isdigit():
      return False
  return True
# ------------------------------ CLEVER SOLUTION ----------------------------- #
def is_isogram(string):
    return len(string) == len(set(string.lower()))
# ----------------------------------- TEST ----------------------------------- #

test.assert_equals(is_isogram("Dermatoglyphics"), True )
test.assert_equals(is_isogram("isogram"), True )
test.assert_equals(is_isogram("aba"), False, "same chars may not be adjacent" )
test.assert_equals(is_isogram("moOse"), False, "same chars may not be same case" )
test.assert_equals(is_isogram("isIsogram"), False )
test.assert_equals(is_isogram(""), True, "an empty string is a valid isogram" )

File: katas/7 kyu/isograms/solution.php
<!-- isograms -->
<?php
use PHPUnit\Framework\TestCase;
/* ------------------------------- MY SOLUTION ------------------------------ */
function isIsogram($string) {
  $string = strtolower($string);
  return $string == "" ? true : count(array_unique(str_split($string , 1)))  == strlen($string) ;
}
/* ---------------------------------- TEST ---------------------------------- */
class MyTestCases extends TestCase
{
    public function testSampleTests() {
      $this->assertEquals(true, isIsogram("Dermatoglyphics"));
      $this->assertEquals(true, isIsogram("isogram"));
      $this->assertEquals(false, isIsogram("aba"), "same chars may not be adjacent");
      $this->assertEquals(false, isIsogram("moOse"), "same chars may not be same case");
      $this->assertEquals(false, isIsogram("isIsogram"));
      $this->assertEquals(true, isIsogram(""), "an empty string is a valid isogram");
    }
}